src/null-safety/migrating-to-null-safety/index.md

```diff
@@ -0,0 +1,435 @@
+---
+title: Migrating to null safety
+description: How to move your existing Dart code to the world of null safety
+---
+
+**TODO: Null safety hasn't shipped yet and this doc still contains some TODOs
+for aspects of null safety (like the shipping stable version number) that aren't
+known yet. Fix those before landing this on master.**
+
+When we decided to add null safety to Dart, we wanted to get as much value out
+of it as possible. That meant designing a fully sound system that guarantees you
+will not see a null reference error. This lets our compilers optimize code
+knowing that `null` cannot flow into certain places. The challenge is that the
+world's existing Dart code is not designed for that level of strictness. How do
+we give users sound null safety without breaking all of the existing Dart code
+in the world?
+
+Our answer is that we let you *migrate* your code to null safety when *you* want
+to. This document explains how that works, and how to get your code into the new
+magical land of null safety.
+
+## The goal
+
+Our goal is that eventually the entire Dart ecosystem will move all of their
+Dart code to null safety. Null unsafe **legacy** Dart code is still supported
```

Comment:
I'm not a big fan of the word 'legacy' as a term for code that doesn't use null safety yet. 'legacy code' often means 'code that is no longer supported', which is not quite where we are yet.

Can we just call it 'Dart code that doesn't yet use null safety is still...' ?

Reply:
"Dart code that doesn't yet use null safety is still" is pretty long. :) Changed it to use "null-unsafe".